.github/workflows/release-quickstart.yml

```diff
@@ -0,0 +1,133 @@
+name: Release Quickstart Workflow
+on:
+  pull_request_target:
+    types:
+      - closed
+    branches:
+      - main
+      - release-v*
+      - release-next
+
+jobs:
+  release-quickstart:
+    name: Release Quickstart Job
+    # when the ref is release-next the label "quickstartrelease" must be present on the PR for this job to run, and when
+    # the ref is a release branch the label is not required
+    if: |
+      github.event.pull_request.merged == true
+      && (
+        github.ref != 'refs/heads/release-next'
+        || contains(github.event.pull_request.labels.*.name, 'quickstartrelease')
```

Comment:
this should be an and not an or?

Reply:
No it has to be an OR for this to work because the AND is the workflow trigger for closed PRs targeting the refs where the qs is released, and this subtracts PRs that were closed but not merged and PRs targeting release-next that were not labeled.

Reply:
i don't understand. it seems to me that it should only happen on a merged, pr, to release-next which has the proper label.

This reads:
do this on a merged pr if the branch is release-next OR the pr contains the quickstartrelease label.

no?

Reply:
It's a confusing set of triggers to be sure, but I couldn't think of another way to express the logic due to the way GitHub works. The resultant logic is: release the quickstart when a PR merges to main, release-v* (hotfix), or release-next (but only if labeled).

This is done by triggering the workflow when any PR closes for those branches, then filtering out the ones that were not merges (like closed "won't do" PRs) and the ones that are targeting release-next but are not labeled.